app/repositories/ContentAPI.scala

```diff
@@ -22,19 +22,43 @@ object ContentAPI {
   private val apiClient = new LiveContentApiClass(Config().capiKey, Config().capiUrl)
   private val previewApiClient = new DraftContentApiClass(Config().capiKey, Config().capiPreviewUrl)
 
-  def countOccurencesOfTagInContents(contentIds: List[String], apiTagId: String): Int= {
+  def countOccurencesOfTagInContents(contentIds: List[String], apiTagId: String): Int = {
+    val builder = StringBuilder.newBuilder
+    var pageSize = 0
+
+    var total = 0
+
```

Comment:
You mean folding into a list of requests? Then processing all those lists after the fact?